src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,263 @@
+
+    self.assertIsNotNone(ppa_id)
+
+    # Notify the SAS UUT about the SAS Test Harness
+    certificate_hash = getCertificateFingerprint(self._sas._tls_config.client_cert)
+    self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,
+                                   'url': self._sas._base_url})
+
+    # As SAS is reset at the beginning of the test, the FAD records should contain
+    # only one zone record containing the PPA that was generated. Hence the first
+    # zone record is retrived and verified if it matches the PPA ID.
+    response = self.TriggerFullActivityDumpAndWaitUntilComplete(
+        config['serverCert'], config['serverKey'])
+    self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+    ppa_zone_data = []
+
+    # Download 'zone' dump file. 
+    for dump_file in response['files']:
+      self.assertContainsRequiredFields("ActivityDumpFile.schema.json", dump_file)
+      downloaded_file = None
+      if dump_file['recordType'] == 'zone':
+        url = str(dump_file['url']).strip()
+        downloaded_file = self._sas.DownloadFile(url)
+        ppa_zone_data.append(downloaded_file)
+
+    # Check if the retrieved FAD PPA zone record matches the PPA ID that was generated
+    # using the admin API.
+    #self.assertEqual(ppa_zone_data[0]['id'], ppa_id)
+
+    uut_ppa_geometry = ppa_zone_data[0]['zone']['features'][0]['geometry']
+    uut_ppa_shapely_geometry = utils.GeoJsonToShapelyGeometry(uut_ppa_geometry)
+
+    # Configure the Census Tract Directory that the PPA uses
+    ppa.ConfigureCensusTractDriver(CONFIG.GetCensusTractsDir())
+
+    # PpaCreationModel requires the input registrationRequests to have 'installationParam'.
+    # But this parameter is removed for devices where conditionals are pre-loaded.
+    # Adding the 'installationParam' into registrationRequests by taking the
+    # corresponding values from conditionalRegistrationData.
+    for device in config['registrationRequests']:
+      if not device.has_key('installationParam'):
+        for conditional_params in config['conditionalRegistrationData'][
+          'registrationData']:
+          # Check if FCC_ID+Serial_Number present in registrationRequest
+          # and conditional_params match and add the 'installationParam'.
+          if conditional_params['fccId'] == device['fccId'] and \
+                          conditional_params['cbsdSerialNumber'] == device['cbsdSerialNumber']:
+            device.update({'installationParam': 
+              conditional_params['installationParam']})
+            device.update({'cbsdCategory': 
+                            conditional_params['cbsdCategory']})
+    # Trigger PPA creation.
+    test_harness_ppa_geometry = ppa.PpaCreationModel(
+        config['registrationRequests'], pal_records)
+    
+    # Convert PPA reference GeoJSON format to Shapely geometry object.
+    test_harness_ppa_shapely_geometry = utils.GeoJsonToShapelyGeometry(
```

Comment:
It could be:
```
test_harness_ppa_shapely_geometry = utils.ToShapely(test_harness_ppa_geometry)  # no need of the json.loads
```
But in fact this line is useless. See below other comments. You dont need to convert to shapely, and can directly call with the geojson the Polygon comparison routine